### PR TITLE
fix: Support optional customize functions returning null case

### PR DESCRIPTION
**Pull request by liamcho**
Fixes: [CLNP-2851](https://sendbird.atlassian.net/browse/CLNP-2851)

### Changelogs
- Fixed a bug where injecting an optional property with null value not rendering the expected default component

[CLNP-2851]: https://sendbird.atlassian.net/browse/CLNP-2851?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bang9** [comment]
TODO:

- [ ] src/modules/ChannelSettings/components/ChannelSettingsUI/index.tsx
- [ ] src/modules/Channel/components/ChannelUI/index.tsx 
- [ ] src/modules/Channel/components/MessageList/index.tsx
- [ ] src/modules/GroupChannel/components/Message/MessageView.tsx
- [ ] src/modules/GroupChannel/components/MessageList/index.tsx
- [ ] src/ui/MessageContent/index.tsx
- [ ] src/ui/MessageItemMenu/index.tsx

File: src/modules/GroupChannelList/context/GroupChannelListProvider.tsx
Comment (bang9) [inline src/modules/GroupChannelList/context/GroupChannelListProvider.tsx:1]: It doesn't seem to be the target that needs modification

File: src/ui/FeedbackIconButton/index.tsx
Comment (bang9) [inline src/ui/FeedbackIconButton/index.tsx:1]: It doesn't seem to be the target that needs modification

File: src/ui/IconButton/index.tsx
Comment (bang9) [inline src/ui/IconButton/index.tsx:1]: It doesn't seem to be the target that needs modification

File: src/ui/ImageRenderer/index.tsx
Comment (bang9) [inline src/ui/ImageRenderer/index.tsx:1]: It doesn't seem to be the target that needs modification

File: src/ui/MessageInput/index.tsx
Comment (bang9) [inline src/ui/MessageInput/index.tsx:1]: It doesn't seem to be the target that needs modification

File: src/ui/Modal/index.tsx
Comment (bang9) [inline src/ui/Modal/index.tsx:1]: It doesn't seem to be the target that needs modification

File: src/ui/ReactionBadge/index.tsx
Comment (bang9) [inline src/ui/ReactionBadge/index.tsx:1]: It doesn't seem to be the target that needs modification

File: src/ui/Toggle/ToggleUI.tsx
Comment (bang9) [inline src/ui/Toggle/ToggleUI.tsx:1]: It doesn't seem to be the target that needs modification

File: src/modules/ChannelSettings/components/LeaveChannel/index.tsx
Comment (bang9) [inline src/modules/ChannelSettings/components/LeaveChannel/index.tsx:1]: It doesn't seem to be the target that needs modification

File: src/modules/GroupChannel/components/Message/MessageView.tsx [hunk 1]
```diff
@@ -146,7 +147,7 @@ const MessageView = (props: MessageViewProps) => {
     animatedMessageId,
     onMessageAnimated,
     usedInLegacy = true,
-  } = props;
+  } = deleteNullish(props);
```
Comment (bang9) [inline src/modules/GroupChannel/components/Message/MessageView.tsx:150]: @liamcho @HoonBaek @AhyoungRyu

When `renderXXX` props are mixed with regular props like this, would it be better to separate them?
Have concerned about potential side effects in the logic when customers pass `null` and it changes to `undefined`. (For example, when business logic checks conditions using `=== null`.)

```tsx
// AS-IS
const {
  propA,
  propB,
  renderPropC = () => <DefaultComponent />
} = deleteNullish(props);

// TO-BE
const { propA, propB } = props;
const { renderPropC = () => <DefaultComponent /> } = deleteNullish(props);
```

Comment (liamcho) [reply src/modules/GroupChannel/components/Message/MessageView.tsx:150]: Its the same no? propA, propB are references so they could be removed after deleteNullish(props)

**Review by liamcho — COMMENTED** [review]
@bang9 can you commit directly and merge while im away?

File: src/modules/GroupChannel/components/Message/MessageView.tsx (hunk 1 above)
Comment (bang9) [reply src/modules/GroupChannel/components/Message/MessageView.tsx:150]: If we are distinguishing between undefined and null for propA and propB in our business logic, the outcome will differ. (The customer passed null, but it is handled as undefined)

**Review by HoonBaek — APPROVED** [review]
LGTM

Is it intentional that `deleteNullish` is separated in some parts and not in others?

**Comment by bang9** [comment]
> LGTM
> 
> Is it intentional that `deleteNullish` is separated in some parts and not in others?

Yes, you can read more -> https://github.com/sendbird/sendbird-uikit-react/pull/1047#discussion_r1558840216

**Pull request merged**